planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp

```diff
@@ -90,11 +90,24 @@ PullOverModule::PullOverModule(
   // for collision check with objects
   vehicle_footprint_ = createVehicleFootprint(vehicle_info_);
 
-  // timer callback for generating candidate paths
-  const auto period_ns = rclcpp::Rate(1.0).period();
-  timer_cb_group_ = node.create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
-  timer_ = rclcpp::create_timer(
-    &node, clock_, period_ns, std::bind(&PullOverModule::onTimer, this), timer_cb_group_);
+  // timer callback for generating lane parking candidate paths
+  const auto lane_parking_period_ns = rclcpp::Rate(1.0).period();
+  lane_parking_timer_cb_group_ =
+    node.create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  lane_parking_timer_ = rclcpp::create_timer(
+    &node, clock_, lane_parking_period_ns, std::bind(&PullOverModule::onTimer, this),
+    lane_parking_timer_cb_group_);
+
+  // freespace parking
+  if (parameters_.enable_freespace_parking) {
+    freespace_planner_ = std::make_unique<FreespacePullOver>(node, parameters, vehicle_info_);
+    const auto freespace_parking_period_ns = rclcpp::Rate(1.0).period();
+    freespace_parking_timer_cb_group_ =
```

Comment:
What is the reason of making separate cb_groups with MutuallyExclusive?

Reply:
[tier iv internal slack](https://star4.slack.com/archives/C03QW0GU6P7/p1677216635243619?thread_ts=1677212370.556509&cid=C03QW0GU6P7)

Both lane and freespace path generations take time to generate, so run them in separate threads.